### ISSUE TITLE
Disable rebuilds for stack, git, build pack changes using image resource 

### ISSUE THREAD

**Issue by doddisam** [buildpacks-community/kpack#1629]
Hi Team,

Is there a way to disable disable image rebuilds for stack, git, build pack changes using image resource in kpack?

**Comment by chenbh** [comment]
I'm not sure why you would use the `Image` resource then, for one-off builds you should use the [`Build` resource](https://github.com/buildpacks-community/kpack/blob/main/docs/build.md) directly.

**Comment by doddisam** [comment]
The reason why i want to use Image is to add k8's service bindings for dependencies and certificates. will build resource allow us to add k8 service bindings ?

**Comment by doddisam** [comment]
Also we want use builder resource not builder image reference as we manage kpack builder. Basically We want our user to use builder (managed by us) not use builder image reference. So wanted to check if we can restrict rebuilds for stack, git and build pack changes using image resource

**Comment by doddisam** [comment]
Can anybody help on above ?

**Comment by tomkennedy513** [comment]
Build has its own mechanism for referencing service bindings. 

There is a spec.services field that you can use like this 

```
services:
- name: settings-xml
  kind: Secret
```

**Comment by doddisam** [comment]
Yes services work for image resource but for for build resource. Below is my image vs build resource. My original questions will service bindings works for build resource ?
**Build resource**
apiVersion: kpack.io/v1alpha2
kind: Build
metadata:
  name: sample-build2
spec:
  tags:
  - harbor.xxx.xxx/pcf-images/korifi/build-test:latest
  serviceAccountName: image-registry-credentials
  builder:
    image: harbor.xxx.xxx/pcf-images/paketobuildpacks/builder-jammy-base:latest
  build:
    env:
      - name: BP_EMBED_CERTS
        value: "true"
    services:
    - name: dependencies
      kind: Secret
      apiVersion: v1
    - name: xxx-chain-certs
      kind: Secret
      apiVersion: v1
  source:
    blob:
      url: https://dcsartifacts.xxxx.xxxx:443/artifactory/dicf/Prisma/cf-sample-sb-0.0.1-SNAPSHOT.jar 

**Image resource.**
apiVersion: kpack.io/v1alpha2
kind: Image
metadata:
  name: demo-java-airgapped-r1-4
  namespace: kpack-testing
spec:
  tag: harbor.xxxx.xxxx/pcf-images/korifi/java-demo-cert-r1-4 # specify target image registry path here
  imageTaggingStrategy: BuildNumber
  serviceAccountName: image-registry-credentials # be sure to use this service account if you push images to harbor.demo.kube-plus.cloud
  builder:
    name: cf-kpack-builder-test
    kind: Builder
  source:
    blob: # this example uses app source code from a public github repository
     url: https://dcsartifacts.xxxx.xxxx:443/artifactory/dicf/Prisma/cf-sample-sb-0.0.1-SNAPSHOT.jar
    # revision: 82cb521d636b282340378d80a6307a08e3d4a4c4
  build:
    env:
      - name: BP_EMBED_CERTS
        value: "true"
  bindings:
  - name: dependencies
    kind: Secret
    apiVersion: v1
  - name: xxx-chain-certs
    kind: Secret
    apiVersion: v1

**Comment by tomkennedy513** [comment]
> We want our users to use builder resource not builder.image reference as we manage kpack builder. so we have a use case to check if we can restrict automatic rebuilds for stack, git and build pack changes using image resource.
> 
> Also even if we use "builder.image" in build resource, can we use k8's service bindings for dependencies ? The reason i am asking this is we have a air gapped env and currently we are downloading dependencies to our artifactory and creating a secret for all mappings. is there a way for us to provide those dependency mappings to build resource ?
> 
> I also explored to mount secret as volume using build resource, looks like that is also not supported.

This should now be possible thanks to #1857
